src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -107,4 +111,43 @@ public ActionRuleDTO createActionRule(
         createdActionRule.getId());
     return createdActionRule;
   }
+
+  public ActionRuleDTO updateActionRule(
+      final UUID id,
+      final String name,
+      final String description,
+      final OffsetDateTime triggerDateTime,
+      final int priority)
+      throws RestClientException {
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getActionSvc().getActionRulesPath() + "/" + id.toString(), null, id);
+
+    final ActionRulePutRequestDTO actionRulePutRequestDTO = new ActionRulePutRequestDTO();
+    actionRulePutRequestDTO.setName(name);
+    actionRulePutRequestDTO.setDescription(description);
+    actionRulePutRequestDTO.setTriggerDateTime(triggerDateTime);
+    actionRulePutRequestDTO.setPriority(priority);
+
+    final HttpEntity<ActionRulePutRequestDTO> httpEntity =
+        restUtility.createHttpEntity(actionRulePutRequestDTO);
+    return restTemplate
+        .exchange(uriComponents.toUri(), HttpMethod.PUT, httpEntity, ActionRuleDTO.class)
+        .getBody();
+  }
+
+  @Override
+  public List<ActionRuleDTO> getActionRulesForActionPlan(final UUID actionPlanId) {
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(
+            appConfig.getActionSvc().getActionRulesForActionPlanPath(), null, actionPlanId);
+
+    return restTemplate
+        .exchange(
+            uriComponents.toUri(),
+            HttpMethod.GET,
+            null,
+            new ParameterizedTypeReference<List<ActionRuleDTO>>() {})
+        .getBody();
```

Comment:
you should check the HTTP status is success before you call getBody... you might get a 404 or a 5XX in which case this code will NPE

Reply:
there will be a HttpClientErrorException thrown if it's 404, but the code which uses this is expecting null to be returned

Reply:
👍